magma/circuit.py

```diff
@@ -62,13 +62,55 @@ class DefinitionContext:
     def __init__(self, placer):
         self.placer = placer
         self._inline_verilog = []
+        self._displays = []
+        self._insert_default_log_level = False
+        self._files = []
+
+    def add_file(self, file):
+        self._files.append(file)
+
+    def finalize_file_opens(self):
+        for file in self._files:
+            self.add_inline_verilog(
+                f"""
+integer \_file_{file.filename} ;
```

Comment:
I've found that these long inline multiline strings make the code harder to read and scan, can we make the private globals at the top of the file? So like

```python
_logger = root_logger()

_VERILOG_FILE_CLOSE = """
integer \_file_{filename} ;
initial \_file_{filename} = $fopen(\"{filename}\", \"{mode}\");
"""

...

            self.add_inline_verilog(_VERILOG_FILE_CLOSE.format(filename=file.filename,
                                                               mode=file.mode)
                                   {}, {})
```